### PR TITLE
Fix select credential bug

### PR DESCRIPTION
**Pull request by nitro-neal**
fixes - https://github.com/TBD54566975/web5-js/issues/447

### CODE REVIEW

File: packages/credentials/tests/presentation-exchange.spec.ts [hunk 1]
```diff
@@ -58,6 +65,24 @@ describe('PresentationExchange', () => {
       expect(actualSelectedVcJwts).to.deep.equal([btcCredentialJwt]);
     });
 
+    it('should return 0 verifiable credentials when they dont match the pd', async () => {
```
Comment (frankhinek) [inline packages/credentials/tests/presentation-exchange.spec.ts:68]: minor nit but probably doesn't hurt to spell out:
```suggestion
    it('should return 0 verifiable credentials when they dont match the presentation definition', async () => {
```

File: packages/credentials/src/presentation-exchange.ts [hunk 2]
```diff
@@ -32,7 +32,13 @@ export class PresentationExchange {
   }): string[] {
     this.resetPex();
     const selectResults: SelectResults = this.pex.selectFrom(presentationDefinition, vcJwts);
-    return Array.from(new Set(selectResults.verifiableCredential as string[] ?? []));
+
+    // If errros the credentials provided didn't satisfy the requirement defined presentationDefinition object
```
Comment (frankhinek) [inline packages/credentials/src/presentation-exchange.ts:36]: Small typo but possible to provide a better explanation based on this detail: https://github.com/TBD54566975/web5-js/issues/447
```suggestion
    // If errros the credentials provided didn't satisfy the requirement defined presentationDefinition object
```

**Review by frankhinek — APPROVED** [review]

File: packages/credentials/src/presentation-exchange.ts [hunk 3]
```diff
@@ -32,6 +32,12 @@ export class PresentationExchange {
   }): string[] {
     this.resetPex();
     const selectResults: SelectResults = this.pex.selectFrom(presentationDefinition, vcJwts);
+
+    // If errors exist in the results object the credentials provided didn't satisfy the requirements in the Presentation Definition
+    if(selectResults.errors?.length) {
```
Comment (diehuxx) [inline packages/credentials/src/presentation-exchange.ts:37]: super nit: despite using JS for years i still get confused about falsy values. do you mind doing `!== 0`?

File: packages/credentials/tests/presentation-exchange.spec.ts [hunk 4]
```diff
@@ -50,6 +50,13 @@ describe('PresentationExchange', () => {
       });
     });
 
+    it('should evaluate credentials without any errors or warnings', async () => {
+      PresentationExchange.satisfiesPresentationDefinition({
```
Comment (diehuxx) [inline packages/credentials/tests/presentation-exchange.spec.ts:54]: nit: make the test description a little more descriptive.
is this test just testing that it doesn't throw as long as the cred satisfies the PD or is it meant to show that it never throws even when the cred fails to satisfy the PD?

**Review by diehuxx — APPROVED** [review]

**Pull request merged**